3.32.1: tagging during maven build fails

### Describe the bug

When building a project with the following properties set in the `pom.xml`:
```xml
<quarkus.container-image.build>true</quarkus.container-image.build>
<quarkus.container-image.group>${project.groupId}</quarkus.container-image.group>
<quarkus.container-image.name>${project.artifactId}</quarkus.container-image.name>
<quarkus.container-image.tag>latest</quarkus.container-image.tag>
<quarkus.container-image.additional-tags>jvm</quarkus.container-image.additional-tags>
```

it seems that re-tagging the image fails:
```
...
[INFO] --- quarkus:3.32.1:build (default) @ quarkus-camel-xml-streaming ---
[INFO] [io.quarkus.container.image.docker.common.deployment.CommonProcessor] Starting (local) container image build for jar using docker
[INFO] [io.quarkus.container.image.docker.common.deployment.CommonProcessor] Executing the following command to build image: 'docker build -f /home/runner/work/quarkus-camel-xml-streaming/quarkus-camel-xml-streaming/src/main/container/Containerfile.temurin -t de.turing85/quarkus-camel-xml-streaming:latest /home/runner/work/quarkus-camel-xml-streaming/quarkus-camel-xml-streaming'
...
[INFO] [io.quarkus.container.image.docker.deployment.DockerProcessor] Built container image de.turing85/quarkus-camel-xml-streaming:latest

[INFO] [io.quarkus.container.image.docker.common.deployment.CommonProcessor] Running 'docker tag de.turing85/quarkus-camel-xml-streaming:latest de.turing85/quarkus-camel-xml-streaming:jvm'
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  36.105 s
[INFO] Finished at: 2026-02-25T21:21:42Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.32.1:build (default) on project quarkus-camel-xml-streaming: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
Error: [ERROR] 	[error]: Build step io.quarkus.container.image.docker.deployment.DockerProcessor#dockerBuildFromJar threw an exception: java.lang.ClassCastException: class io.quarkus.bootstrap.json.JsonNull cannot be cast to class io.quarkus.bootstrap.json.JsonObject (io.quarkus.bootstrap.json.JsonNull and io.quarkus.bootstrap.json.JsonObject are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @2fce8243)
[ERROR] 	at io.quarkus.container.image.docker.common.deployment.CommonProcessor.determineBuiltContainerInfo(CommonProcessor.java:152)
[ERROR] 	at io.quarkus.container.image.docker.common.deployment.CommonProcessor.buildFromJar(CommonProcessor.java:98)
[ERROR] 	at io.quarkus.container.image.docker.deployment.DockerProcessor.dockerBuildFromJar(DockerProcessor.java:69)
[ERROR] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:898)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:242)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1474)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```

(full build log can be found [here (`github.com`)](https://github.com/turing85/quarkus-camel-xml-streaming/actions/runs/22416042281/job/64903176230) and [here (`github.com`)](https://github.com/turing85/quarkus-xa/actions/runs/22416673597/job/64904271536))

### Expected behavior

Image is tagged successfully, and build succeeds.

### Actual behavior

Build fails with above exception.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.32.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

`Apache Maven 3.9.8 (36645f6c9b5079805ea5009217e36f2cffd34256)`

### Additional information

_No response_

--- a/extensions/container-image/container-image-docker-common/deployment/src/main/java/io/quarkus/container/image/docker/common/deployment/CommonProcessor.java
+++ b/extensions/container-image/container-image-docker-common/deployment/src/main/java/io/quarkus/container/image/docker/common/deployment/CommonProcessor.java
@@ -145,12 +145,16 @@ public abstract class CommonProcessor<C extends CommonConfig> {
                 });
             }
 
-            String workingDir = ((JsonString) config.get("WorkingDir")).value();
+            String workingDir = null;
+            JsonValue workingDirVal = config.get("WorkingDir");
+            if (workingDirVal instanceof JsonString js) {
+                workingDir = js.value();
+            }
 
             Optional<String> baseImage = Optional.empty();
 
-            JsonObject labels = config.get("Labels");
-            if (labels != null) {
+            JsonValue labelsVal = config.get("Labels");
+            if (labelsVal instanceof JsonObject labels) {
                 JsonValue baseNameLabelObj = labels.get("org.opencontainers.image.base.name");
                 if (baseNameLabelObj instanceof JsonString s) {
                     baseImage = Optional.of(s.value());

